sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleLastValueAggregator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import io.opentelemetry.sdk.metrics.data.MetricData.DoublePoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import java.util.Map;
+
+public final class DoubleLastValueAggregator extends AbstractAggregator {
+
+  private static final double DEFAULT_VALUE = 0.0;
+  private static final AggregatorFactory AGGREGATOR_FACTORY =
+      new AggregatorFactory() {
+        @Override
+        public Aggregator getAggregator() {
+          return new DoubleLastValueAggregator();
+        }
+      };
+
+  private final AtomicDouble current = new AtomicDouble(DEFAULT_VALUE);
+
+  public static AggregatorFactory getFactory() {
+    return AGGREGATOR_FACTORY;
+  }
+
+  @Override
+  void doMergeAndReset(Aggregator aggregator) {
+    DoubleLastValueAggregator other = (DoubleLastValueAggregator) aggregator;
+    other.current.set(this.current.getAndSet(DEFAULT_VALUE));
```

Comment:
Are we willing to accept that we might be overwriting something that was recorded later than what we're currently holding onto? I'm thinking of the dimensionality reduction use-case for this.

Reply:
This is use only for observer so all the events are recorded once in the callback execution, if we want to support this for CounterInstruments or MeasureInstruments then we need a time as well and preserver the ordering.

Reply:
We should make that very clear in the documentation of this class, then, and perhaps in the class name. Can we just call the class `ObserverLastValue` ?

Reply:
Documented in the description this limitation. For the moment I would not rename it since we most likely are going to fix this.